packages/firebase_core/android/src/main/java/io/flutter/plugins/firebase/core/FlutterFirebaseAppRegistrar.java

```diff
@@ -7,12 +7,10 @@
 import java.util.List;
 
 public class FlutterFirebaseAppRegistrar implements ComponentRegistrar {
-  private static final String LIBRARY_NAME = "flutter-fire-core";
-  private static final String LIBRARY_VERSION = "0.4.0+3";
-
   @Override
   public List<Component<?>> getComponents() {
     return Collections.<Component<?>>singletonList(
-        LibraryVersionComponent.create(LIBRARY_NAME, LIBRARY_VERSION));
+        LibraryVersionComponent.create(
+            "flutter-fire-core", BuildConfig.VERSION_NAME.replaceAll("\\+", "-")));
```

Comment:
Does this replace all need to be done at runtime - can we move the logic to build time inside of the `flutterFireVersion()` method?